docs/dxp/7.x/en/installation-and-upgrades/setting-up-liferay-dxp/tuning-your-jvm.md

```diff
@@ -0,0 +1,103 @@
+
+Here are the JVM tuning topics:
+
+* [Garbage Collector](#garbage-collector)
+* [Code Cache](#code-cache)
+* [Java Heap](#java-heap)
+* [JVM Advanced Options](#jvm-advanced-options)
+
+Garbage collection is first.
+
+## Garbage Collector
+
+Choosing the appropriate garbage collector (GC) helps improve the responsiveness of your Liferay DXP deployment. Use the concurrent low pause collectors:
+
+```bash
+-XX:+UseParNewGC -XX:ParallelGCThreads=16 -XX:+UseConcMarkSweepGC
+-XX:+CMSParallelRemarkEnabled -XX:+CMSCompactWhenClearAllSoftRefs
+-XX:CMSInitiatingOccupancyFraction=85 -XX:+CMSScavengeBeforeRemark
+```
+
+You may choose from other available GC algorithms, including parallel throughput collectors and G1 collectors. Start tuning using parallel collectors in the new generation and concurrent mark sweep (CMS) in the old generation.
+
+**Note:** the `ParallelGCThreads` value (e.g., `ParallelGCThreads=16`) varies  based on the type of CPUs available. Set the value according to CPU specification. On Linux machines, report the number of available CPUs by running `cat /proc/cpuinfo`.
+
+**Note:** There are additional "new" algorithms like G1, but Liferay Engineering's tests for G1 indicated that it does not improve performance. Since your application performance may vary, you should add G1 to your testing and tuning plans.
+
+## Code Cache
+
+Java's just-in-time (JIT) compiler generates native code to improve performance. The default size is `48m`. This may not be sufficient for larger applications. Too small a code cache reduces performance, as the JIT isn't able to optimize high frequency methods. For Liferay DXP,  start with `64m` for the initial code cache size.
+
+```bash
+-XX:InitialCodeCacheSize=64m -XX:ReservedCodeCacheSize=96m
+```
+
+Examine the efficacy of the parameter changes by adding the following logging parameters:
+
+```bash
+-XX:+PrintCodeCache -XX:+PrintCodeCacheOnCompilation
+```
+
+## Java Heap
+
+When most people think about tuning the Java memory heap, they think of setting the maximum and minimum memory of the heap. Unfortunately, most deployments require far more sophisticated heap tuning to obtain optimal performance, including tuning the young generation size, tenuring durations, survivor spaces, and many other JVM internals.
+
+For most systems, it's best to start with at least the following memory settings:
+
+```bash
+server -XX:NewSize=700m -XX:MaxNewSize=700m -Xms2560m -Xmx2560m -XX:MetaspaceSize=512m
+-XX:MaxMetaspaceSize=512m -XX:SurvivorRatio=6 -XX:TargetSurvivorRatio=9 -XX:MaxTenuringThreshold=15
+```
+
+On systems that require large heap sizes (e.g., above 4GB), it may be beneficial to use large page sizes. You can activate large page sizes like this:
+
+```bash
+-XX:+UseLargePages -XX:LargePageSizeInBytes=256m
```

Comment:
@jhinkey Just for reference, in Liferay benchmark we use `-XX:LargePageSizeInBytes=2m`.